python/packages/autogen-agentchat/src/autogen_agentchat/agents/_openai_assistant_agent.py

```diff
@@ -0,0 +1,503 @@
+    The agent maintains a thread of conversation and can use various tools including:
+    - Code interpreter: For executing code and working with files
+    - File search: For searching through uploaded documents
+    - Custom functions: For extending capabilities with user-defined tools
+
+    Key Features:
+    - Supports multiple file formats including code, documents, images
+    - Can handle up to 128 tools per assistant
+    - Maintains conversation context in threads
+    - Supports file uploads for code interpreter and search
+    - Vector store integration for efficient file search
+    - Automatic file parsing and embedding
+
+    Example:
+        .. code-block:: python
+
+            from openai import AsyncClient
+            from autogen_agentchat.agents import OpenAIAssistantAgent
+
+            # Create an OpenAI client
+            client = AsyncClient(api_key="your-api-key", base_url="your-base-url")
+
+            # Create an assistant with code interpreter
+            assistant = OpenAIAssistantAgent(
+                name="Python Helper",
+                description="Helps with Python programming",
+                client=client,
+                model="gpt-4",
+                instructions="You are a helpful Python programming assistant.",
+                tools=[{"type": "code_interpreter"}],
+            )
+
+            # Upload files for the assistant to use
+            await assistant.on_upload_for_code_interpreter("data.csv", cancellation_token)
+
+            # Get response from the assistant
+            response = await assistant.on_messages(
+                [TextMessage(source="user", content="Analyze the data in data.csv")], cancellation_token
+            )
+
+            # Clean up resources
+            await assistant.delete_uploaded_files(cancellation_token)
+            await assistant.delete_assistant(cancellation_token)
+
+    Args:
+        name (str): Name of the assistant
+        description (str): Description of the assistant's purpose
+        client (AsyncClient): OpenAI API client instance
+        model (str): Model to use (e.g. "gpt-4")
+        instructions (str): System instructions for the assistant
+        tools (Optional[Iterable[CodeInterpreterToolParam | FileSearchToolParam | BaseToolParam[ArgsT, ReturnT]]]): Tools the assistant can use
+        assistant_id (Optional[str]): ID of existing assistant to use
+        metadata (Optional[object]): Additional metadata for the assistant
+        response_format (Optional[AssistantResponseFormatOptionParam]): Response format settings
+        temperature (Optional[float]): Temperature for response generation
+        tool_resources (Optional[ToolResources]): Additional tool configuration
+        top_p (Optional[float]): Top p sampling parameter
+    """
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        client: AsyncClient,
+        model: str,
+        instructions: str,
+        tools: Optional[
```

Comment:
For custom functions, the input to the api is {"type": "function", "function": {function-json-schema}. What I did was create a wrapper around `autogen_core.components.Tool` to keep it consistent. In my test, we pass tools like:

```
    tools = [
        {"type": "code_interpreter"},
        {"type": "file_search"},
        {"type": "tool", "tool": DisplayQuizTool()},
    ]
```

What I could do is modify the input to be Literal["code_interpreter", "file_search]|Tool rather than enforce the api format in the input and then we just reformat the params internally.

Reply:
Then I think we can use the approach from `AssistantAgent` more directly.

Reply:
Yep. We can do the conversion internally.

Reply:
Updated, now it should be working in a similar way to `AssistantAgent` and the built-in tools can be passed directly as strings.